apis/apiextensions/v1/composition_transforms.go

```diff
@@ -137,6 +137,8 @@ func (m *MathTransform) Resolve(input any) (any, error) {
 		return *m.Multiply * i, nil
 	case int:
 		return *m.Multiply * int64(i), nil
+	case float64:
+		return float64(*m.Multiply) * i, nil
```

Comment:
Given that actual inputs are never float64 (Kubernetes encoding requires they always be int64s), is this intended to cover the case in which `input` has been converted to a float64 in a previous transform?

Reply:
Is this actually guaranteed by the parser or just a frontend validation?

Given that the JSON spec does not distinguish between integers and floats and only defines an abstract `number` type (like in Javascript). This means in Go, all numbers that are parsed with `json.Unmarshall` (which is used by k8s-yaml internally) are converted to float64 by default if the type is not known, like in `Unstructured` that is used for XRs and XRCs.

See https://go.dev/play/p/0CI40vKXcnc